src/fides/api/ops/models/privacy_preference.py

```diff
@@ -0,0 +1,206 @@
+
+from enum import Enum
+from typing import Any, Type
+
+from sqlalchemy import ARRAY, Column
+from sqlalchemy import Enum as EnumColumn
+from sqlalchemy import ForeignKey, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict, MutableList
+from sqlalchemy.orm import Session, relationship
+from sqlalchemy_utils import StringEncryptedType
+from sqlalchemy_utils.types.encrypted.encrypted_type import AesGcmEngine
+
+from fides.api.ops.common_exceptions import PrivacyNoticeHistoryNotFound
+from fides.api.ops.db.base_class import JSONTypeOverride
+from fides.api.ops.models.privacy_notice import (
+    PrivacyNotice,
+    PrivacyNoticeHistory,
+    PrivacyNoticeRegion,
+)
+from fides.api.ops.models.privacy_request import PrivacyRequest, ProvidedIdentity
+from fides.core.config import CONFIG
+from fides.lib.db.base import Base  # type: ignore[attr-defined]
+
+
+class UserConsentPreference(Enum):
+    opt_in = "opt_in"  # The user wants to opt in to the notice
+    opt_out = "opt_out"  # The user wants to opt out of the notice
+    acknowledge = "acknowledge"  # The user has acknowledged this notice
+
+
+class RequestOrigin(Enum):
+    privacy_center = "privacy_center"
+    privacy_modal = "privacy_modal"
+
+
+class PrivacyPreferenceHistory(Base):
+    """The DB ORM model for storing PrivacyPreferenceHistory, used for saving
+    every time consent preferences are saved for reporting purposes.
+
+    add fides_user_provided_identity nullable FK
+    add fides_user_device_id encrypted field
+    add ip address
+    add consent preference method?
+    """
+
+    affected_system_status = Column(
+        MutableDict.as_mutable(JSONB), server_default="{}", default=dict
+    )
+    email = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+        nullable=True,
+    )
+    phone_number = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+        nullable=True,
+    )
+    preference = Column(EnumColumn(UserConsentPreference), nullable=False, index=True)
+    privacy_notice_history_id = Column(
+        String, ForeignKey(PrivacyNoticeHistory.id), nullable=False, index=True
+    )
+    privacy_request_id = Column(
+        String, ForeignKey(PrivacyRequest.id), nullable=True, index=True
+    )
+    provided_identity_id = Column(
+        String,
+        ForeignKey(ProvidedIdentity.id),
+        nullable=True,
+    )
+    relevant_systems = Column(
+        MutableList.as_mutable(ARRAY(String))
+    )  # Systems whose data use match.  This doesn't necessarily mean we propagate. Some may be intentionally skipped later.
+    request_origin = Column(EnumColumn(RequestOrigin))
```

Comment:
Just missing some cleanup, thanks for this.

`nullable=True` is the default so if I'm missing it, I probably knew it wasn't required for the onset. Some where I've added nullable=True explicitly, it's possible I changed them from False->True, as I originally had marked too many fields as non-nullable, or I copied a field off another model.

I think it's easier to parse if only the fields that are required have nullable=False, otherwise that argument is left off altogether.